src/Infrastructure/Persistence/Configuration/TodoItemConfiguration.cs

```diff
@@ -12,10 +12,11 @@ public class TodoItemConfiguration : IEntityTypeConfiguration<TodoItem>
     public void Configure(EntityTypeBuilder<TodoItem> builder)
     {
         builder.HasKey(t => t.Id);
+
         builder.Property(t => t.Id)
             .HasConversion(x => x.Value,
                 x => new TodoItemId(x))
-            .ValueGeneratedOnAdd();
+            .ValueGeneratedNever(); // TODO DM: Discuss this with SAW team
```

Comment:
Would be keen to get feedback on this.  This is required if we want to manually set GUIDs used for IDs  manually in code.

Reply:
I think it's very rarely you'd want to manually set GUIDs for an entry.

Setting GUID manually is usually needed for application layer logic (I'm thinking of use case where the ID come from external e.g. Auth0). In this case, we just need to create a new field to track this ID, so our domain object entry has the `Id` controlled by our application, and another ID e.g. `auth0Id` coming from external.

That's my opinion - keen to hear other's thoughts 😄

Reply:
@christoment fair point.  I've reverted this change, so that we err on the side of simplicity for the template.  Individual projects can always decide to go whichever way they need.